libs/libc/stdio/Kconfig

```diff
@@ -53,7 +53,7 @@ config NUNGET_CHARS
 
 config LIBC_FLOATINGPOINT
 	bool "Enable floating point in printf"
-	default n
+	default !DEFAULT_SMALL && ARCH_FPU
```

Comment:
Why we rely on `ARCH_FPU`? The `LIBC_FLOATINGPOINT` does not have anything about hard vs soft FPU. Or for archs with `ARCH_FPU` the code size increase is low?